comit/src/lnd.rs

```diff
@@ -248,27 +258,44 @@ impl WaitForAccepted for LndConnectorAsSender {
     async fn wait_for_accepted(&self, params: &Params) -> anyhow::Result<Accepted> {
         // No validation of the parameters because once the payment has been
         // sent the sender cannot cancel it.
-        while self
-            .find_payment(params, PaymentStatus::InFlight)
-            .await?
-            .is_none()
-        {
+        loop {
             tokio::time::delay_for(Duration::from_millis(self.retry_interval_ms)).await;
-        }
 
-        Ok(Accepted)
+            let payment = match self.find_payment(params).await? {
+                Some(payment) => payment,
+                None => continue,
+            };
+
+            match payment.status {
```

Comment:
How is this necessary/related to this patch?

Reply:
The problem is, that we cannot "replay" halight protocols the same we do for other protocols because `lnd` only exposes the latest state.

So we need to actually explicitly match, which states are acceptable, taking into account that the desired state is already in the past, i.e. when looking for `Open`, `Accepted` is also good :)

Reply:
To do that reliably, we needed to decouple fetching the invoice from matching against the desired state.